subsys/lorawan/lw_priv.c

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2020 Manivannan Sadhasivam <mani@kernel.org>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <zephyr.h>
+
+#include "lw_priv.h"
+
+#include <LoRaMac.h>
+
+static const char *const status2str[] = {
+	[LORAMAC_STATUS_OK] = "OK",
+	[LORAMAC_STATUS_BUSY] = "Busy",
+	[LORAMAC_STATUS_SERVICE_UNKNOWN] = "Service unknown",
+	[LORAMAC_STATUS_PARAMETER_INVALID] = "Parameter invalid",
+	[LORAMAC_STATUS_FREQUENCY_INVALID] = "Frequency invalid",
+	[LORAMAC_STATUS_DATARATE_INVALID] = "Datarate invalid",
+	[LORAMAC_STATUS_FREQ_AND_DR_INVALID] = "Frequency or datarate invalid",
+	[LORAMAC_STATUS_NO_NETWORK_JOINED] = "No network joined",
+	[LORAMAC_STATUS_LENGTH_ERROR] = "Length error",
+	[LORAMAC_STATUS_REGION_NOT_SUPPORTED] = "Region not supported",
+	[LORAMAC_STATUS_SKIPPED_APP_DATA] = "Skipped APP data",
+	[LORAMAC_STATUS_DUTYCYCLE_RESTRICTED] = "Duty-cycle restricted",
+	[LORAMAC_STATUS_NO_CHANNEL_FOUND] = "No channel found",
+	[LORAMAC_STATUS_NO_FREE_CHANNEL_FOUND] = "No free channel found",
+	[LORAMAC_STATUS_BUSY_BEACON_RESERVED_TIME] =
+		"Busy beacon reserved time",
+	[LORAMAC_STATUS_BUSY_PING_SLOT_WINDOW_TIME] =
+		"Busy ping-slot window time",
+	[LORAMAC_STATUS_BUSY_UPLINK_COLLISION] = "Busy uplink collision",
+	[LORAMAC_STATUS_CRYPTO_ERROR] = "Crypto error",
+	[LORAMAC_STATUS_FCNT_HANDLER_ERROR] = "FCnt handler error",
+	[LORAMAC_STATUS_MAC_COMMAD_ERROR] = "MAC command error",
+	[LORAMAC_STATUS_CLASS_B_ERROR] = "ClassB error",
+	[LORAMAC_STATUS_CONFIRM_QUEUE_ERROR] = "Confirm queue error",
+	[LORAMAC_STATUS_MC_GROUP_UNDEFINED] = "Multicast group undefined",
+	[LORAMAC_STATUS_ERROR] = "Unknown error",
+};
+
+const char *lorawan_status2str(int status)
+{
+	if (status < ARRAY_SIZE(status2str)) {
```

Comment:
Since status is a signed type, it might make sense to also ensure that `status >= 0` or to use an unsigned type for status. (also for the other similar functions here `lorawan_X2Y`)

Reply:
Done!